sympy/combinatorics/rewritingsystem.py

```diff
@@ -374,14 +373,14 @@ def _add_to_automaton(self, rules):
             # Add dead states
             if not rule in self.reduction_automaton.states:
                 self.reduction_automaton.add_state(rule, state_type='d', rh_rule=all_rules[rule])
-
+        automaton_alphabet = set(automaton_alphabet)
+        # Add accept states
+        for rule in all_rules:
```

Comment:
Why do you need this to be in a separate loop?

Reply:
I posted it on the channel, the assertion error caused in the test cases might have been because of this. It works fine if we add the accept states after all the dead states are added. There must've been an overlap in the case with the assertion error

Reply:
Okay, I see. So there must be an overlap between the dead states and the accept states (is that what you meant on the channel?), and whenever both are possible we should be making them dead - because then we actually need to make a substitution and that's important. Is it possible to rewrite the state type? Because then you could have everything in one loop but run `self.reduction_automaton.add_state(rule, state_type='d', rh_rule=all_rules[rule])` even if `rule` is already in `self.reduction_automaton.states` (as an `accept` state) - maybe you could modify `add_state` to rewrite a state with new information in that case?